src/gflags.cc

```diff
@@ -800,7 +826,7 @@ CommandLineFlag* FlagRegistry::SplitArgumentLocked(const char* arg,
                                     kError, key->c_str());
       return NULL;
     }
-    if (strcmp(flag->type_name(), "bool") != 0) {
+    if (!flag->OfType<bool>()) {
```

Comment:
I think it is absolutely fine to have it public. And public here anyway only means public internally, not as part of the public API. In any case, the type enumeration is there to avoid the use of actual template types or subclasses for each particular flag type. Thus, the enumeration should be part of the public interface to query the flag's type rather than `TypeName`, which should only be used for the human readable help output, conversion of enum value to string.